### PR TITLE
Fix for Scispark-180: Docker file for a SciSpark dev environment 

### PR DESCRIPTION
**Pull request by sujen1412**
I have an automated build linked to my repository, I can change it to master here once we review this. The build exists at https://hub.docker.com/r/sujenshah/scispark/

### CODE REVIEW

**Comment by sujen1412** [comment]
@BrianWilson1 this is the docker. You can test it locally and let me know if there are any changes to be made. Thanks.

**Comment by kwhitehall** [comment]
@sujen1412  add readme to the folder for the first timers to docker plz

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<h1>Docker</h1>
+
+Docker is an open-source project that automates the deployment of Linux applications inside software containers.
+Learn more about docker containers [here](https://www.docker.com/).
+
+<h2> Requirements </h2>
+
+Instal Docker following the guidelines for your system [here](https://docs.docker.com/engine/installation/)
+
+<h2> SciSpark Docker </h2>
```
Comment (kwhitehall) [inline docker/README.md]: please update to SciSpark dev Docker

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+<h2> Requirements </h2>
+
+Instal Docker following the guidelines for your system [here](https://docs.docker.com/engine/installation/)
+
+<h2> SciSpark Docker </h2> 
+To run the SciSpark docker follow the below steps: 
+
+* Make sure you have docker installed
+* Now, ```$ git clone https://github.com/SciSpark/SciSpark.git```
+* run ```$ cd SciSaprk/docker/```
```
Comment (kwhitehall) [inline docker/README.md]: typo. SciSpark/docker

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+* If you want to build the docker container from scratch run 
+
+```$ docker build -t scispark -f Dockerfile .```
+
+else you can pull and existing image from docker hub by running 
+
+```$ docker pull sujenshah/scispark```
```
Comment (kwhitehall) [inline docker/README.md]: please push this existing image to a repo on SciSpark and update this link there.

File: docker/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+* After the build completed successfully, run the docker by executing this command 
+
+```docker run -it scispark /bin/bash``` 
+
+or if you have pulled from docker hub run 
+   
+```docker run -it sujenshah/scispark /bin/bash``` 
+* The above command will open up a shell with root access in the scispark docker
```
Comment (kwhitehall) [inline docker/README.md:31]: same here.

File: docker/README.md (hunk 3 above)
Comment (sujen1412) [reply docker/README.md]: What do you mean by push existing image to a repo on Scispark ? Do we have an account under SciSpark on docker hub ?

**Comment by kwhitehall** [comment]
Looks like the fail is due to the check not being able to pull the jars to build the env for the 2.11, 2.0 environment. @rahulpalamuttam  can you please investigate Travis
@sujen1412  given the nature of this pr, ie. a docker file and README I'm okay with merging.
EDIT
Any objections?

**Comment by rahulpalamuttam** [comment]
Not sure why some builds are getting HTTP errors for sbt and others aren't.
It's not spark/scala version specific either. In some cases a different version combo fails.

I'm able to go to the sites that travis says returned HTTP error responses for and pull the jars just fine.
Definitely a mystery.

**Comment by kwhitehall** [comment]
Ack'ed. Saw an open issue on travis repo. Check into the Gitter room for more details.

**Pull request merged**